### PR TITLE
Add proof verification into the tests.

### PR DESCRIPTION
**Pull request by shimkiv**
#474

### CODE REVIEW

File: tests/artifacts/javascript/on-chain-state-mgmt-zkapp-ui.js [hunk 1]
```diff
@@ -96,7 +93,12 @@ updateButton.addEventListener('click', async (event) => {
       zkAppInstance.sign(zkAppPrivateKey);
     });
 
-    await transaction.prove();
+    const proof = await transaction.prove();
+
+    if (verificationKey) {
+      verify(proof[0], verificationKey.data);
```
[the remark this answers is not in the document]
Comment (shimkiv) [reply tests/artifacts/javascript/on-chain-state-mgmt-zkapp-ui.js:99]: Forgotten about the `Promise` in return, thank you!

**Review by MartinMinkov — APPROVED** [review]
Looks great, thanks! :rocket:

**Comment by shimkiv** [comment]
@mitschabaude @MartinMinkov 
Weird that tests still fail by:

```
Cannot read properties of undefined (reading 'proof')
```

Am I missing something?

**Comment by MartinMinkov** [comment]
You are storing the result of `Mina.transaction` into `deploymentTransaction` but then calling `transaction.prove` and `transaction.send()`. Maybe it should be `deploymentTransaction` instead?

**Comment by mitschabaude** [comment]
Not sure what causes the error you mention @shimkiv! Another thing: To make verification work in the browser, you probably have to merge `releases` since that is where I pushed the fix

**Comment by mitschabaude** [comment]
@shimkiv I think you're signing the tx, that's why no proof is created, so `proofs[0]` is undefined`

File: tests/artifacts/javascript/on-chain-state-mgmt-zkapp-ui.js [hunk 2]
```diff
@@ -96,7 +93,13 @@ updateButton.addEventListener('click', async (event) => {
       zkAppInstance.sign(zkAppPrivateKey);
```
Comment (mitschabaude) [inline tests/artifacts/javascript/on-chain-state-mgmt-zkapp-ui.js:93]: this causes it to create no proof

Comment (shimkiv) [reply tests/artifacts/javascript/on-chain-state-mgmt-zkapp-ui.js]: Thank you!
What was the driver behind this decision (if sign present no proofs will be generated)?
Is it documented somewhere?

Comment (mitschabaude) [reply tests/artifacts/javascript/on-chain-state-mgmt-zkapp-ui.js]: not documented, but planned to be!
sign means a signature is to be used as authorization. 'signature' and 'proof'  are two mutually exclusive forms of authorization. you can't have both.

Comment (shimkiv) [reply tests/artifacts/javascript/on-chain-state-mgmt-zkapp-ui.js]: Got it, thank you!

**Pull request merged**